packages/jsdoc-plugins/lib/relation-fixer/addmissingdoclets.js

```diff
@@ -0,0 +1,145 @@
+		const docletsToAdd = getDocletsToAdd( childDoclet, relation, options );
+		const newDoclets = [];
+
+		docletsToAdd.forEach( d => {
+			const cloned = cloneDeep( d );
+			let docletsWithSameLongname = [];
+
+			cloned.longname = getNewLongname( d, childDoclet );
+			cloned.memberof = childDoclet.longname;
+
+			// Add property `inherited`
+			if ( checkIfAddInheritedProperty( childDoclet, d, relation ) ) {
+				cloned.inherited = true;
+			}
+
+			docletsWithSameLongname = data.filter( d => d.longname === cloned.longname );
+
+			if ( docletsWithSameLongname.length === 0 ) {
+				// If there was no doclet for that member, simply add it to new doclets.
+				newDoclets.push( cloned );
+			} else if ( checkExplicitlyInherits( docletsWithSameLongname ) ) {
+				// If doclet for that member already existed and used `inheritdoc` or`overrides`.
+				// Add `ignore` property to existing doclets.
+				docletsWithSameLongname.forEach( d => {
+					d.ignore = true;
+				} );
+
+				newDoclets.push( cloned );
+			}
+		} );
+
+		return newDoclets;
+	}
+
+	// Gets doclets from entities related to current doclet ( e.g. implemented by it ) and matching criteria given in options.
+	function getDocletsToAdd( doclet, relation, options = {} ) {
+		if ( !isNonEmptyArray( doclet[ relation ] ) ) {
+			return [];
+		}
+
+		const parents = doclet[ relation ];
+		const docletsToAdd = [];
+		parents.forEach( ln => {
+			const toAdd = data.filter( d => {
+				let optionalCondition = true;
+
+				for ( const key of Object.keys( options ) ) {
+					if ( d[ key ] !== options[ key ] ) {
+						optionalCondition = false;
+					}
+				}
+
+				return d.memberof === ln && optionalCondition;
+			} );
+
+			docletsToAdd.push( ...toAdd );
+		} );
+
+		return docletsToAdd;
+	}
+
+	// Checks if memberDoclet was inherited from a parent class.
+	function checkIfAddInheritedProperty( childDoclet, memberDoclet, relation ) {
+		if ( relation === 'augmentsNested' ) {
+			return true;
+		}
+
+		const memberDocletParent = data.find( d => d.longname === memberDoclet.memberof );
+		let result = false;
+
+		if ( isNonEmptyArray( memberDocletParent.descendants ) ) {
+			memberDocletParent.descendants.forEach( ln => {
+				const doclet = data.find( d => d.longname === ln );
+
+				if ( doclet.kind === 'class' ) {
+					if ( isNonEmptyArray( doclet.descendants ) &&
+						doclet.descendants.indexOf( childDoclet.longname ) !== -1 ) {
+						result = true;
+					}
+				}
+			} );
+		}
+
+		return result;
+	}
+
+	entitiesWhichNeedNewDoclets.forEach( d => {
+		// Add missing static members from interfaces.
+		newDoclets.push( ...getNewDoclets( d, 'implementsNested', {
+			scope: 'static'
+		} ) );
+		// Add missing static members from extended classes.
+		newDoclets.push( ...getNewDoclets( d, 'augmentsNested', {
+			scope: 'static'
+		} ) );
+	} );
+
+	data.push( ...newDoclets );
+
+	return data;
+};
+
+function getNewLongname( parentDoclet, childDoclet ) {
```

Comment:
Agree. It's unclear what's "new". In case of problems with describing it clearly you can also write a short doc what it does.